internal/dag/builder_test.go

```diff
@@ -83,29 +83,91 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 						MatchLabels: map[string]string{
 							"app": "contour",
 						},
+						MatchExpressions: []metav1.LabelSelectorRequirement{{
+							Key:      "type",
+							Operator: "In",
+							Values:   []string{"controller"},
+						}},
 					},
 				},
 			}},
 		},
 	}
 
+	gatewayNoSelector := &gatewayapi_v1alpha1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gatewayWithSelector",
+			Namespace: "default",
+		},
+		Spec: gatewayapi_v1alpha1.GatewaySpec{
+			Listeners: []gatewayapi_v1alpha1.Listener{{
+				Port:     80,
+				Protocol: "HTTP",
+			}},
+		},
+	}
+
 	tests := map[string]struct {
 		objs                         []interface{}
 		disablePermitInsecure        bool
 		fallbackCertificateName      string
 		fallbackCertificateNamespace string
+		gateway                      *gatewayapi_v1alpha1.Gateway
 		want                         []Vertex
 	}{
 		"insert basic single route, single hostname": {
+			gateway: gatewayWithSelector,
 			objs: []interface{}{
-				gateway,
 				kuardService,
 				&gatewayapi_v1alpha1.HTTPRoute{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "basic",
 						Namespace: "default",
 						Labels: map[string]string{
-							"app": "contour",
+							"app":  "contour",
```

Comment:
let's add a test case where the route's labels don't match the gateway's selector and verify it doesn't get included

Reply:
👍